core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -181,6 +181,19 @@ URI uri() {
         }
     }
 
+    Cookies cookies() {
+        final List<String> cookieStrings = getAll(HttpHeaderNames.COOKIE);
+        if (cookieStrings.isEmpty()) {
+            return Cookies.of();
+        }
+        return Cookie.fromCookieHeaders(cookieStrings);
+    }
+
+    Cookies setCookies() {
+        final List<String> cookieHeaders = getAll(HttpHeaderNames.SET_COOKIE);
+        return Cookie.fromSetCookieHeaders(cookieHeaders);
+    }
```

Comment:
This is a bit confusing to me. `cookies()` only get overridden by subclass and `setCookies()` used as a utility method. 🤔

Reply:
At first I implement cookies in `Default{Response, Request}Headers`. Then I realized builder of these classes need same method. So I move cookie implementation to `HttpHeaderBase`. I know it is bit confusing, but at the same time, also want to name the method which returns set-cookie header as 'cookies`. 🥺 Can you suggest a better way to achieve this?

Reply:
I'm leaning toward using `cookie{s}` for both `{Request,Response}Headers`. Since many application frameworks offer high-level abstraction API for cookies, some users don't distinguish between `Cookie` and `Set-Cookie` header.
In my case, the following API design looks good to me for simplicity.
```java
RequestHeaders.builder()
              .cookie(...);
RequestHeaders requestHeaders = ...;
Cookies cookies = requestHeaders.cookies();

ResponseHeaders.builder()
               .cookie(...);
ResponseHeaders responseHeaders = ...;
Cookies cookies = responseHeaders.cookies();
```

Let's wait for other folks' opinions. (+ Sorry for the misdirection. 😅)

Reply:
Looks good to me. 👍

Reply:
In `AbstractHttpRequestBuilder`, there are setters which name is `cookie` that takes one cookie, and `cookies` that take `Iterable<? extends Cookie>`. I think it would be better to provide the same interface to handle cookie. How do you think about it?

Reply:
Agreed. It must be nice we provide the same interface like `AbstractHttpRequestBuilder`.